xinference/core/event.py

```diff
@@ -34,11 +34,21 @@ class Event(TypedDict):
     event_content: str
 
 
+class FIFOQueue(asyncio.Queue):
+    def __init__(self, maxsize=0):
+        super().__init__(maxsize)
+
+    async def put(self, item):
+        if self.full():
+            await self.get()  # 丢弃最旧的项目
+        await super().put(item)
+
+
 class EventCollectorActor(xo.StatelessActor):
     def __init__(self):
         super().__init__()
-        self._model_uid_to_events: Dict[str, queue.Queue] = defaultdict(  # type: ignore
-            lambda: queue.Queue(maxsize=MAX_EVENT_COUNT_PER_MODEL)
```

Comment:
How about using a fixed-size deque instead of queue.Queue?

Reply:
> How about using a fixed-size deque instead of queue.Queue?

Using deque instead of queue.Queue is a better approach. I've updated the file and tested it, and everything is functioning smoothly.